File: Longest_Common_Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.size()==1)
        {
            return strs[0];
        }
        
        else 
        {
      sort(strs.begin(),strs.end());
       int l = strs.size()-1;
        int min_length = min(strs[0].length(),strs[l].length());
        
        string left = strs[0];
        string right = strs[l];
       int i=0;
        while(i <min_length && left[i] == right[i])
              i++;
        
        
        string pre = left.substr(0,i);
        return pre;
            
        }
        return "";
        
        
    }
};

File: longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs):
        if (len(strs)==0):
            return ""
        if (len(strs)==1):
            return strs[0]

        for j in xrange(0,len(strs[0])):
            for i in xrange(0,len(strs)-1):
                if (j>=len(strs[i+1]) or strs[i][j]!=strs[i+1][j]):
                    if j==0:
                        return ""
                    else:
                        return strs[i][0:j]
        return strs[0]
